react-app-drawer-component/src/app-drawer.jsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+
+class AppDrawer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      menuClicked: false,
+      iconClicked: false
```

Comment:
I think you can simplify your code a lot by just having one state property only like `isOpen`.